tests/plugins/discovery/shell_discovery_plugin/test_shell_discovery_plugin.py

```diff
@@ -0,0 +1,155 @@
+"""Unit tests for the Shell discovery plugin."""
+import contextlib
+import os
+import subprocess
+
+import mock
+from yapsy.PluginManager import PluginManager
+
+import statick_tool
+from statick_tool.discovery_plugin import DiscoveryPlugin
+from statick_tool.exceptions import Exceptions
+from statick_tool.package import Package
+from statick_tool.plugins.discovery.shell_discovery_plugin import ShellDiscoveryPlugin
+
+
+# From https://stackoverflow.com/questions/2059482/shell-temporarily-modify-the-current-processs-environment
+@contextlib.contextmanager
+def modified_environ(*remove, **update):
```

Comment:
it might make sense (as a separate issue) to extract this function into a statick_tool_test library/file. I believe it's used in a few different test files.